### DISCUSSION TITLE
请问是否有办法设置不收集评论者的IP、UA和所属地信息？

### DISCUSSION THREAD

**Discussion by 493490561** [walinejs/waline#1315] [Q&A] [answered]
我已经通过设置[环境变量](https://waline.js.org/reference/server.html#%E7%8E%AF%E5%A2%83%E5%8F%98%E9%87%8F)使用户UA和归属地不显示，但后台仍然可以看到这些信息，我不想收集他们的隐私信息。希望得到解答，谢谢！

**Comment by lizheming** [comment, marked as answer]
可以自定义钩子实现 https://waline.js.org/reference/server.html#presave-comment

```js
// index.js 
const Waline = require('@waline/vercel');
module.exports = Waline({
  preSave(data) {
    delete data.ua;
    delete data.ip;
  }
});

**Reply by 493490561** [reply]
太感谢了！！
